lamindb/models/feature.py

```diff
@@ -777,28 +777,117 @@ def __init__(
                 )
 
     @classmethod
-    def from_df(cls, df: pd.DataFrame, field: FieldAttr | None = None) -> SQLRecordList:
-        """Create Feature records for columns."""
+    def from_df(
+        cls, df: pd.DataFrame, field: FieldAttr | None = None, *, mute: bool = True
+    ) -> SQLRecordList:
+        """Create Feature records for dataframe columns.
+
+        Args:
+            df: Source DataFrame to extract column information from
+            field: FieldAttr for Feature model validation, defaults to Feature.name
+            mute: Whether to mute Feature creation similar names found warnings
+        """
         field = Feature.name if field is None else field
         registry = field.field.model  # type: ignore
         if registry != Feature:
             raise ValueError("field must be a Feature FieldAttr!")
+
         categoricals = categoricals_from_df(df)
         dtypes = {}
         for name, col in df.items():
             if name in categoricals:
                 dtypes[name] = "cat"
             else:
                 dtypes[name] = serialize_pandas_dtype(col.dtype)
-        with logger.mute():  # silence the warning "loaded record with exact same name "
-            features = [
-                Feature(name=name, dtype=dtype) for name, dtype in dtypes.items()
-            ]  # type: ignore
+        if mute:
+            with logger.mute():
+                features = [
+                    Feature(name=key, dtype=dtype) for key, dtype in dtypes.items()
+                ]  # type: ignore
+        else:
+            features = [Feature(name=key, dtype=dtype) for key, dtype in dtypes.items()]  # type: ignore
```

Comment:
The fact that the key logic gets duplicated because of `with logger.mute()` isn't super nice

I suggest rather than duplicating core logic to toggle logger verbosity depending on the state of `mute`

https://github.com/laminlabs/lamin-utils/blob/2a2c54522565f8d043686c247868718dfb6fb0da/lamin_utils/_logger.py#L262